cameo/strain_design/deterministic/linear_programming.py

```diff
@@ -426,3 +430,394 @@ def _repr_html_(self):
         </ul>
         %s""" % (self._target, self.data_frame._repr_html_())
         return html_string
+
+
+class GrowthCouplingPotential(StrainDesignMethod):
+    """
+    Optimize the Growth Coupling Potential through a combination of knockouts, knockins and medium additions
+
+    Use the 'run' method to optimize the MILP problem and find the best solution.
+
+    In order to find multiple sub-optimal solutions through the use of solution pools, please refer to the
+    relevant solvers documentation. The underlying solver object can be accessed through
+        GrowthCouplingPotential(...).model.solver.problem
+
+
+    Parameters
+    ----------
+    model : cobra.Model
+        A cobra model object containing the target reaction as well as all native and heterologous reactions.
+    target : cobra.Reaction or string
+        The reaction (or reaction ID) that is to be growth-coupled
+    knockout_reactions : list
+        A list of ID's for the reactions that the algorithm should attempt to knock out
+    knockin_reactions : list
+        A List of ID's for the reactions that are heterologous and that the algorithm should attempt to add
+    medium_additions : list
+        A list of metabolite ID's for the the metabolites that should potentially be added to the medium
+    n_knockouts : int
+        The maximum allowed number of knockouts
+    n_knockins : int
+        The maximum allowed number of knockins
+    n_medium : int
+        The maximum allowed number of medium additions
+    use_solution_pool : Bool
+        Set to True when using the Gurobi solver to identify multiple alternative solutions at once
+    """
+    def __init__(
+            self, model, target, knockout_reactions, knockin_reactions, medium_additions,
+            n_knockouts, n_knockin, n_medium, remove_blocked=True, use_solution_pool=False, **kwargs
+    ):
+        super(GrowthCouplingPotential, self).__init__(**kwargs)
+
+        # Check model for forced fluxes, which should not be present. The zero solution must be allowed.
+        for r in model.reactions:
+            if r.lower_bound > 0 or r.upper_bound < 0:
+                raise ValueError(
+                    "%s has a forced flux. Remove the reaction or change its bounds for the function to work." % r.id)
+
+        # If a cobra.Reaction is given as target, convert to its ID string.
+        if isinstance(target, cobra.core.Reaction):
+            target = target.id
+        self.target = target
+
+        if target not in knockout_reactions:
+            knockout_reactions.append(target)
+
+        self.original_model = model
+        self.biomass_id = [r for r in model.reactions if r.objective_coefficient != 0][0].id
+        self.model = model = model.copy()
```

Comment:
Assigning to `self.model` seems to have no effect since another model copy is assigned on line 691.
```suggestion
        model = model.copy()
```